sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -844,6 +858,11 @@ public PartitionState apply(TopicPartition tp) {
           partitionStates.get(i).nextOffset = ckptMark.getNextOffset();
         }
       }
+
+      messagesConsumedPerSplit = Metrics.counter("io.splits", source.id + ".messagesConsumed");
+      bytesConsumedPerSplit = Metrics.counter("io.splits", source.id + ".bytesConsumed");
+      backlogBytesPerSplit = Metrics.gauge("io.splits", source.id + ".backlog.bytes");
+      backlogMessagesPerSplit = Metrics.gauge("io.splits", source.id + ".backlog.messages");
```

Comment:
I see. I might add these later. Just a simple stop watch around the calls might not be good enough anyway (e.g. `availableRecordsQueue.put()` might block for a very long time if the pipeline is stuck or is extremely slow).